invenio_app_ils/ui/backoffice/src/pages/ItemDetails/ItemDetailsContainer.js

```diff
@@ -0,0 +1,32 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { ItemDetails } from './components/ItemDetails/ItemDetails';
+
+export default class ItemDetailsContainer extends Component {
+  constructor(props) {
+    super(props);
+    this.fetchItemDetails = this.props.fetchItemDetails;
+  }
+
+  componentDidMount() {
```

Comment:
This could be another HOC probably

Reply:
IMHO, I would go for a HOC component that implements the logic of fetching and instead of hiding the rendering of the page component inside it's render function I would render `this.props.children` and pass the page components explicitely in the template that this container is used. Just a thought about future PRs :)

Reply:
@zzacharo, @ntarocco this is a very good idea, may I suggest to create a separate issue so we can merge this one and keep the re-architecture of data fetching isolated?